packages/fuels-test-helpers/src/node.rs

```diff
@@ -271,4 +298,32 @@ pub async fn server_health_check(client: &FuelClient) {
     if !healthy {
         panic!("error: Could not connect to fuel core server.")
     }
+    true
+}
+
+pub fn get_socket_address() -> SocketAddr {
+    let free_port = pick_unused_port().expect("No ports free");
+    SocketAddr::new("127.0.0.1".parse().unwrap(), free_port)
+}
+
+pub struct FuelService {
+    pub bound_address: SocketAddr,
+}
+
+impl FuelService {
+    pub async fn new_node(config: Config) -> Result<Self, AnyError> {
+        let bound_address = match config {
+            node_config if node_config.addr.port() != 0 && is_free(node_config.addr.port()) => {
+                node_config.addr
+            }
+            node_config if !is_free(node_config.addr.port()) => {
+                panic!("Error: Address already in use")
```

Comment:
Why not just `bail!` since you're already returning a Result? Let the decision to panic, if it is made, be higher up.